pkg/tls/marshal.go

```diff
@@ -126,3 +159,27 @@ func ParsePrivateKey(b []byte) (*rsa.PrivateKey, error) {
 
 	return nil, errors.New("failed to parse private key")
 }
+
+// ParseCertificateChain takes slice/chain of the certificates
+// and required certificate type and return certificate from the chain
+// TODO: Needs implementation
+func ParseCertificateChain(certs []*x509.Certificate, certType CertificateType) (*x509.Certificate, error) {
+	// TODO: make dynamic certificate reconstruction here
+	return certs[len(certs)-1], nil
+}
+
+// UniqueCert takes slice of the certificate and
+// returns certificate slice with unique values
+func UniqueCert(certs []*x509.Certificate) []*x509.Certificate {
+	seen := make(map[string]struct{}, len(certs))
+	j := 0
+	for _, v := range certs {
+		if _, ok := seen[string(v.Raw)]; ok {
```

Comment:
Not sure if this is the right way to sort unique certs `string(v.Raw)`

Reply:
works for me